totp/totp.py

```diff
@@ -45,4 +60,6 @@ def check_code(secret: str, code: str,
     """
     if len(code) != 4:
         raise ValueError("TOTP code must be 4 digits string, is '{}' instead".format(code))
-    return generate_code(secret, seconds_since_the_epoch) == code
+
+    return blake3(generate_code(secret, seconds_since_the_epoch).encode('utf-8')).digest() == blake3(code.encode('utf-8')).digest()
```

Comment:
this line makes no sense.
`blake3(data1) == blake3(data2)` is for all practical purposes equal to `data1 == data2`